File: other_language/c++/dynamic_array.cpp
#include <iostream>
using namespace std;

int main()
{
        /*
            static allocation which means the size of array must
            be constant
            what if we want to take size of array from user?
            will create dynamic allocation by using pointer

            int n;
            int *arr = new int[n];

        */

        const int n = 5;
        int arr[n];
        int sum = 0;

        for (int i = 0; i < n; i++)
        {
            arr[i] = (i + 1) * 10;
            sum += arr[i];
        }

        cout<<"sum: "<<sum;

}

File: other_language/c#/dynamic_array.cs
using System;
class HelloWorld
{
  static void Main()
  {
    int n, sum= 0;

    // take n from user
    Console.Write("enter a number: ");
    n = int.Parse(Console.ReadLine());
      
    //we not have a pointer in c# so we use []
    int []arr = new int[n];
    
    for (int i = 0; i < n; i++)
    {
        arr[i] = (i + 1) * 10;
        sum += arr[i];
    }
    
    Console.Write("sum: " + sum);
      
  }
}
 
